File: SelfHomeApp/app/src/main/java/it/dani/selfhomeapp/middle/ipersistence/IDevicePersistence.java
package it.dani.selfhomeapp.middle.ipersistence;

import androidx.annotation.NonNull;

import java.util.Set;

public interface IDevicePersistence {
    public @NonNull String freeText(@NonNull String text);
    public @NonNull boolean toggleDevice(@NonNull String text);
    public Set<String> getDevices();
    public boolean setStateDevice(@NonNull String deviceName, int state);
    public int getDeviceState(@NonNull String deviceName);
}

File: SelfHomeApp/app/src/main/java/it/dani/selfhomeapp/middle/ipersistence/IDevicePersistence.kt
package it.dani.selfhomeapp.middle.ipersistence

interface IDevicePersistence {
    fun freeText(text : String) : String
    fun toggleDevice(deviceName : String) : Boolean
    fun getDevices() : Set<String>
    fun setStateDevice(deviceName : String, state : Int) : Boolean
    fun getDeviceState(deviceName : String) : Int
}